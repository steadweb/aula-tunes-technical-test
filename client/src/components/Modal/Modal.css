.modal {
  display: none;
  position: fixed;
  border: 4px solid var(--secondary-color);
  min-height: 200px;
  width: 600px;
  left: 50%;
  margin-left: -300px;
  background-color: var(--primary-color);
  top: 50%;
  margin-top: -100px;
  z-index: 2;
}

.modal__image {
  display: flex;
  align-self: center;
}

.modal__content {
  align-self: center;
  width: 100%;
}

.modal__overlay {
  position: fixed;
  width: 100%;
  height: 100%;
  background: var(--primary-color);
  z-index: 1;
  top: 0;
  opacity: 0.6;
  display: none;
}

.modal__content .copyright {
  font-size: 10px;
}

.modal__content .genre {
  display: inline-block;
  margin: 5px;
  border: 1px solid var(--secondary-color);
  border-radius: 15px;
  font-size: 11px;
  padding: 4px 12px;
  cursor: pointer;
}

.modal__content .genre:hover {
  background-color: var(--secondary-color);
  color: var(--primary-color);
}

@media only screen and (max-width: 600px) {
  .modal {
    width: calc(100% - 60px);
    padding: 10px;
    margin: 0 20px;
    left: 0;
    top: 50px;
    flex-direction: column;
  }
}
