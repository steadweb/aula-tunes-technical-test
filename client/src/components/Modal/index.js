import React from 'react';
import './Modal.css';

function Modal({
  artistName,
  name,
  releaseDate,
  artworkUrl100,
  genres,
  copyright,
  enabled,
  close,
  genreFilter,
}) {
  const style = { display: enabled ? 'flex' : 'none' };

  return (
    <>
      <div className='modal' style={style}>
        <div className='modal__image'>
          <img src={artworkUrl100} alt='Album Cover' />
        </div>
        <div className='modal__content'>
          <h2>{artistName}</h2>
          <h3>
            {name} ({releaseDate})
          </h3>
          <p>
            {genres &&
              genres.map((genre) => (
                <span
                  key={genre.genreId}
                  className='genre'
                  onClick={() => genreFilter(genre)}>
                  {genre.name}
                </span>
              ))}
          </p>
          <p className='copyright'>{copyright}</p>
        </div>
      </div>
      <div className='modal__overlay' style={style} onClick={close}></div>
    </>
  );
}

export default Modal;
