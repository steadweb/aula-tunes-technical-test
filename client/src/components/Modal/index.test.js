import React from 'react';
import renderer from 'react-test-renderer';
import Modal from './index';

describe('Modal', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Modal />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
