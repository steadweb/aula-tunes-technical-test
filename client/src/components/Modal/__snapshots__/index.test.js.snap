// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
Array [
  <div
    className="modal"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="modal__image"
    >
      <img
        alt="Album Cover"
      />
    </div>
    <div
      className="modal__content"
    >
      <h2 />
      <h3>
         (
        )
      </h3>
      <p />
      <p
        className="copyright"
      />
    </div>
  </div>,
  <div
    className="modal__overlay"
    style={
      Object {
        "display": "none",
      }
    }
  />,
]
`;
